lib/woocommerce.js

```diff
@@ -8,65 +8,70 @@ var logger = require('./logger'),
  * @param {object} options:
  * url: the store url with no protocol or trailing forward slash,
  * e.g. mystore.com (required)
- * port: the port number to reach your store's API (default: 80)
  * ssl: true/false, is your api on https or http (default: false)
  * consumerKey: the consumer key generated in the WC control panel (required)
  * secret: the secret generated in the WC control panel (required)
  * apiPath: the path of the api (default: /wc-api/v2)
  * logLevel: 0: error only, 1: error & info
  */
-var WooCommerce = function(options) {
 
-  this.options = options || {};
-
-  if (!this.options.consumerKey || !this.options.secret) {
-    throw new Error('The consumer key and secret are required');
+class WooCommerce {
+  constructor(options) {
+    this.options = options || {};
+
+    if (!this.options.consumerKey || !this.options.secret) {
+      throw new Error('The consumer key and secret are required');
+    }
+
+    if (!this.options.url) {
+      throw new Error('The URL is required');
+    }
+
+    // Set defaults
+    this.options.logLevel = this.options.logLevel || 0;
+    this.options.apiPath = this.options.apiPath || '/wc-api/v2';
+
+    // Automatically set ssl when not set
+    if (typeof options.ssl === 'undefined') {
+      const https = /https/.test(this.options.url);
+      this.options.ssl = https ? true : false;
```

Comment:
All of this could be reduced to `options.ssl = options.ssl || /https/.test(this.options.url);`